rpc-client/src/nonblocking/rpc_client.rs

```diff
@@ -673,24 +667,44 @@ impl RpcClient {
         .await
     }
 
+    async fn send_transaction_and_get_latest_blockhash(
+        &self,
+        transaction: &impl SerializableTransaction,
+        config: Option<RpcSendTransactionConfig>,
+    ) -> ClientResult<(Hash, Signature)> {
+        let (latest_blockhash, signature) = join!(
+            async {
+                if transaction.uses_durable_nonce() {
+                    self.get_latest_blockhash_with_commitment(CommitmentConfig::processed())
+                        .await
+                        .map(|v| v.0)
+                } else {
+                    Ok(*transaction.get_recent_blockhash())
+                }
+            },
+            async {
+                if let Some(config) = config {
+                    self.send_transaction_with_config(transaction, config).await
+                } else {
+                    self.send_transaction(transaction).await
+                }
+            },
```

Comment:
Since both of these branches are awaitable, does Rust let you just drop the future in here without the extra async block?
```suggestion
            if let Some(config) = config {
                self.send_transaction_with_config(transaction, config)
            } else {
                self.send_transaction(transaction)
            },
```

Reply:
```
error[E0308]: `if` and `else` have incompatible types
   --> rpc-client/src/nonblocking/rpc_client.rs:688:17
    |
685 | /             if let Some(config) = config {
686 | |                 self.send_transaction_with_config(transaction, config)
    | |                 ------------------------------------------------------ expected because of this
687 | |             } else {
688 | |                 self.send_transaction(transaction)
    | |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected future, found a different future
689 | |             }
    | |_____________- `if` and `else` have incompatible types
    |
    = note: distinct uses of `impl Trait` result in different opaque types
help: consider `await`ing on both `Future`s
    |
686 ~                 self.send_transaction_with_config(transaction, config).await
687 |             } else {
688 ~                 self.send_transaction(transaction).await
    |
```